pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -323,30 +326,37 @@ func loadIPSecKeys(r io.Reader) (uint8, error) {
 			offset = -1
 		}
 		if spiI > linux_defaults.IPsecMaxKeyVersion {
-			return 0, fmt.Errorf("encryption Key space exhausted, id must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion, s[0])
+			return 0, 0, fmt.Errorf("encryption Key space exhausted, id must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion, s[0])
 		}
 		if spiI == 0 {
-			return 0, fmt.Errorf("zero is not a valid key to disable encryption use `--enable-ipsec=false`, id must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion, s[0])
+			return 0, 0, fmt.Errorf("zero is not a valid key to disable encryption use `--enable-ipsec=false`, id must be nonzero and less than %d. Attempted %q", linux_defaults.IPsecMaxKeyVersion, s[0])
 		}
 		spi = uint8(spiI)
 
-		authkey, err := decodeIPSecKey(s[2+offset])
+		keyLen, authkey, err = decodeIPSecKey(s[2+offset])
 		if err != nil {
-			return 0, fmt.Errorf("unable to decode authkey string %q", s[1+offset])
+			return 0, 0, fmt.Errorf("unable to decode authkey string %q", s[1+offset])
 		}
 		authname := s[1+offset]
 
 		if strings.HasPrefix(authname, "rfc") {
+			icvLen, err := strconv.Atoi(s[3+offset])
+			if err != nil {
+				return 0, 0, fmt.Errorf("ICVLen is invalid or missing")
+			}
+
 			ipSecKey.Aead = &netlink.XfrmStateAlgo{
 				Name:   authname,
 				Key:    authkey,
-				ICVLen: 128,
+				ICVLen: icvLen,
 			}
+			keyLen = icvLen / 8
```

Comment:
I guess that users putting nonsense values in here (eg `123`) will just quietly set a weird ICV len? Is there anything deeper in the stack that might depend on sane values? Should we sanitize these further?

Reply:
There will be an error deeper in the stack but I’ll add something here to give readable errors.